src/Controls/src/Core/HandlerImpl/Window/Window.Windows.cs

```diff
@@ -1,14 +1,41 @@
 ﻿#nullable disable
 using System;
-using System.Collections.Generic;
-using System.Text;
-using Microsoft.Maui.Handlers;
+using Microsoft.Maui.Graphics;
+using Microsoft.UI.Xaml.Controls;
 
 namespace Microsoft.Maui.Controls
 {
 	public partial class Window
 	{
+		Rect[] _titleBarDragRectangles;
 		internal UI.Xaml.Window NativeWindow =>
 			(Handler?.PlatformView as UI.Xaml.Window) ?? throw new InvalidOperationException("Window Handler should have a Window set.");
+
+		void UpdateTitleBarDragRectangles(Rect[] titleBarDragRectangles)
+		{
+			_titleBarDragRectangles = titleBarDragRectangles;
+			Handler?.UpdateValue(nameof(IWindow.TitleBarDragRectangles));
+		}
+
+		static void MapWindowTitle(IWindowHandler handler, IWindow window)
+		{
+			if (window is Window controlsWindow)
+			{
+				handler
+					.PlatformView
+					.UpdateTitle(window, controlsWindow.GetCurrentlyPresentedMauiContext());
+			}
+
+			WindowHandler.MapTitle(handler, window);
```

Comment:
No, that would cause a stackoverflow exception. That would just call back into this overridden mapper.

What I actually want here is for Matt's Modify PR to land and we just use that but for now all this does is forward the call to the original mapper after I've weaved my behavior into the app.

Reply:
Keeping this comment here so we don't forget to do this one my PR lands.